src/display/text_layer.js

```diff
@@ -527,21 +536,55 @@ var renderTextLayer = (function renderTextLayerClosure() {
       this._capability.reject('canceled');
     },
 
+    _processItems(items, styleCache) {
+      for (let i = 0, len = items.length; i < len; i++) {
+        this.textContentItemsStr.push(items[i].str);
+        appendText(this, items[i], styleCache);
+      }
+    },
+
     _render: function TextLayer_render(timeout) {
-      var textItems = this._textContent.items;
-      var textStyles = this._textContent.styles;
-      for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(this, textItems[i], textStyles);
+      let capability = createPromiseCapability();
+      let pump;
```

Comment:
Nit: I don't think that you need to define this variable here, since you're only using it in one branch below.